docs/contributor/arch/021-refactor-performance-tests.md

```diff
@@ -0,0 +1,44 @@
+    - The [FileLogWriter](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/testbed/datasenders/stanza.go) converts OTLP logs to text lines and writes them to a temporary log file. This is not suitable for our Log Agent, since it tails logs which are only written to the stdout.
+    - The workaround in the POC is to write a custom `stdoutLogGenerator` which implements the `DataSender` interface. The `stdoutLogGenerator` is very similar to the `FileLogWriter`, but it writes logs to stdout instead of a temporary log file.
+- The receiver used in the mock backend is [listening](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/testbed/testbed/receivers.go#L81) on `127.0.0.1:4317` (localhost only).
+    - This works fine for the performance tests in the opentelemetry-collector-contrib repo, because these test are executed locally. However, we need to test our Telemetry module collectors in a kubernetes cluster.
+    - Since this [address](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/testbed/testbed/receivers.go#L81) is hardcoded in the receiver and not configurable, the workaround in the POC is to copy over the receiver source code and change the address to `0.0.0.0:4317` (all interfaces).
+- Whenever the API for the Opentelemtry Collector Testbed changes, we will need to update the source code for the [loadgenerator](../pocs/opentelemetry-testbed/loadgenerator/) and the [mockbackend](../pocs/opentelemetry-testbed/mockbackend/) and rebuild their images.
+
+#### Conclusion
+The extra metric (data items sent/received) that we can get from using the components from the Opentelemtry Collector Testbed is not worth all the disadvantages mentioned above. Therefore, we are going to use the already existing components (e.g load generators, mock backend, ...) from our [e2e testkit](../../../test/testkit/).
+
+## Decision
+
+- Rewrite the existing performance tests in Golang.
+- Use the already existing components from our [e2e testkit](../../../test/testkit/).
+- Add tests for the new Log Gateway and Log Agent. They will have similar test cases as the existing ones for Metric Gateway and Metric Agent. The setup for testing the Log Gateway and Log Agent is shown respectively in the diagrams below:
+
+![arch](./../assets/log-gateway-perf-test-setup.svg)
+
+![arch](./../assets/log-agent-perf-test-setup.svg)
+
+- For testing the Log Agent, it would be better to use a JSON log generator in which we can add custom attributes. This will allow us to test the perfromance of the [operators](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver#operators) we use in the filelog receiver in the Log Agent. The operators currently used can be found in the [golden test file](https://github.com/kyma-project/telemetry-manager/blob/main/internal/otelcollector/config/log/agent/testdata/config.yaml#L59) for the Log Agent.  
```

Comment:
```suggestion
- For testing the Log Agent, it would be better to use a JSON log generator in which we can add custom attributes. This will allow us to test the performance of the [operators](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver#operators) we use in the filelog receiver in the Log Agent. To find the operators currently used, see the [golden test file](https://github.com/kyma-project/telemetry-manager/blob/main/internal/otelcollector/config/log/agent/testdata/config.yaml#L59) for the Log Agent.  
```